src/filehandling.c

```diff
@@ -596,7 +596,7 @@ void hc_fclose (HCFILE *fp)
 
     unzClose (fp->ufp);
   }
-  else
+  else if (fp->pfp)
```

Comment:
We should do the same change we do here:

```
@@ -596,7 +596,7 @@ void hc_fclose (HCFILE *fp)
 
     unzClose (fp->ufp);
   }
-  else
+  else if (fp->pfp)
   {
     fclose (fp->pfp);
   }
```

To all functions which follow the same code structure/convention. For instance:

```
@@ -576,7 +576,7 @@ void hc_fflush (HCFILE *fp)
   else if (fp->is_zip)
   {
   }
-  else
+  else if (fp->pfp)
   {
     fflush (fp->pfp);
   }
```

Reply:
True. I was only looking at these from resource leak point of view.

Reply:
Are you going to fix them or want me to merge it like this?